### PR TITLE
CB-17750; reduce the data node count to 2

### PR DESCRIPTION
**Pull request by RokolyA94**
CB-17750; reduce the data node count to 2

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java [hunk 1]
```diff
@@ -22,13 +25,31 @@ public class HdfsRoleConfigProvider extends AbstractRoleConfigProvider {
...
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case HdfsRoles.DATANODE:
-                return List.of(
+                List<ApiClusterTemplateConfig> configs = new ArrayList<>();
+                configs.add(
                         config(FAILED_VOLUMES_TOLERATED, NUM_FAILED_VOLUMES_TOLERATED.toString())
                 );
+                if (entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())
+                && isNamenodeHA(source)) {
```
Comment (lnardai) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:47]: How are we differentiating here that this is a config is only meant for a DL?
Wouldn't this mean that if we have this entitlement we'll have the max replication count set to 2?

**Review by lnardai — CHANGES_REQUESTED** [review]
See comment about DH with Namenode.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java (hunk 1 above)
Comment (kkalvagadda1) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:47]: I agree. This change should be limited to datalake. 
The current change will limit the replication count to 2 for any cluster that had HDFS.

Comment (kkalvagadda1) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:47]: Add the below condition to limit it to the datalake cluster.

source.getStackType().equals(StackType.DATALAKE)

**Review by kkalvagadda1 — CHANGES_REQUESTED** [review]

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java (hunk 1 above)
Comment (RokolyA94) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java]: resolved

**Review by kkalvagadda1 — APPROVED** [review]

**Review by doktoric — APPROVED** [review]

**Review by lnardai — APPROVED** [review]

**Pull request merged**